packages/ra-ui-materialui/src/list/datagrid/Datagrid.tsx

```diff
@@ -259,7 +260,6 @@ export const Datagrid: FC<DatagridProps> = React.forwardRef((props, ref) => {
                                     hasBulkActions,
                                     isRowSelectable,
                                     onSelect,
-                                    resource,
```

Comment:
We no longer need to pass down resource as there is now always a ResourceContext, even in controlled mode.